app/code/Magento/Catalog/Block/Product/View/Attributes.php

```diff
@@ -85,13 +85,15 @@ public function getAdditionalData(array $excludeAttr = [])
 
                 if (!$product->hasData($attribute->getAttributeCode())) {
                     $value = __('N/A');
+                } elseif ($value instanceof Phrase) {
+                    $value = (string)$value;
                 } elseif ((string)$value == '') {
                     $value = __('No');
```

Comment:
This is weird logic. Why empty string should be transformed to `No`. Looks like this some obsolete workaround that should be removed. If this cause any issue then it definitely should be fixed in another way.
Please remove this `elseif` logic as it has no sense.

Reply:
Most likely this was used to display boolean attributes with Yes/No values.

Reply:
@orlangur Yes/No values at this point already handled in `\Magento\Eav\Model\Entity\Attribute\Frontend\AbstractFrontend::getValue`. And this is dangerous to substitute empty value with "No" as empty value may appear for different reasons (e.g. empty text field, dropdown without selected item, etc.). Magento should rid of such code, decrease coupling between different classes and do not base implementation on current execution flow but build reusable components with high cohesion so changes in one system part will not cause bugs in another.